// @framer/rmarinac89.ios-alert/code/Alert.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType } from "framer"
import "./styles.css"

export function Alert(props) {
    const {
        width,
        height,
        appearance,
        title,
        message,
        primary,
        secondary,
        tertiary,
        destructiveColor,
    } = props

    const mode = appearance ? "light" : "dark"
    const rightBorder = secondary && tertiary ? " " : "border-right"
    const colorType = destructiveColor && !tertiary ? "#FF3B30" : "#007AFF"
    const checkforMessage = message ? "2px" : "0px"
    const preferredOrder = tertiary.length > 0 ? 1 : 2
    const secondaryOrder = tertiary.length > 0 ? 2 : 1

    return (
        <Frame
            name="alert-dialog"
            width={width}
            height={height}
            background="transparent"
        >
            <div className={"alert " + mode}>
                <div className="content">
                    <h1
                        className="title"
                        style={{ marginBottom: checkforMessage }}
                    >
                        {title ? title : "A Short Title is Best"}
                    </h1>
                    <p className="message">{message}</p>
                </div>

                <div
                    className={
                        tertiary.length > 0 ? "actions-col" : "actions-row"
                    }
                >
                    {secondary && (
                        <span
                            className={"action " + rightBorder}
                            style={{
                                order: secondaryOrder,
                                width:
                                    tertiary.length === 0 &&
                                    secondary.length > 0
                                        ? "50%"
                                        : "100%",
                            }}
                        >
                            {secondary}
                        </span>
                    )}

                    <span
                        className="preferred-action"
                        style={{
                            order: preferredOrder,
                            color: colorType,
                            width:
                                tertiary.length === 0 && secondary.length > 0
                                    ? "50%"
                                    : "100%",
                        }}
                    >
                        {primary ? primary : "Preffered Action"}
                    </span>

                    {tertiary && (
                        <span
                            className="action"
                            style={{ order: 3, width: "100%" }}
                        >
                            {tertiary}
                        </span>
                    )}
                </div>
            </div>
        </Frame>
    )
}

Alert.defaultProps = {
    height: 143,
    width: 270,
    primary: "OK",
    secondary: "Cancel",
    tertiary: "",
    destructiveColor: false,
}

addPropertyControls(Alert, {
    appearance: {
        type: ControlType.Boolean,
        title: "Mode",
        enabledTitle: "Light",
        disabledTitle: "Dark",
    },
    title: {
        type: ControlType.String,
        title: "Title",
        defaultValue: "A Short Title Is Best",
    },
    message: {
        type: ControlType.String,
        title: "Message",
        defaultValue: "A message should be a short, complete sentence.",
    },
    primary: {
        type: ControlType.String,
        title: "Primary",
    },
    secondary: {
        type: ControlType.String,
        title: "Secondary",
    },
    tertiary: {
        type: ControlType.String,
        title: "Tertiary",
    },
    destructiveColor: {
        type: ControlType.Boolean,
        title: "Destructive",
        enabledTitle: "True",
        disabledTitle: "False",
        hidden(props) {
            return props.tertiary.length > 0
        },
    },
})
